### PR TITLE
 Mckin 9196 - Provide human readable user interactions to new instructor tool

### PR DESCRIPTION
**Pull request by jawad-khan**
https://edx-wiki.atlassian.net/browse/MCKIN-9196

Usage:
New [Instructor tool](https://github.com/edx/edx-platform/blob/d02efb1d2449cc183cb64d4a9c5c7bf58a146fcb/lms/djangoapps/instructor_task/tasks_helper/grades.py#L600-L751) will be able to get all submission data from scorm xblock.

### CODE REVIEW

File: scormxblock/scormxblock.py [hunk 1]
```diff
@@ -437,6 +439,52 @@ def proxy_content(self, request, suffix=''):
...
+            sco_data = sco.get('data') or {}
+            interactions_count = sco_data.get(interaction_prefix + '_count', 0)
+
+            for interaction_index in range(interactions_count):
+                current_interaction_prefix = interaction_prefix + str(interaction_index) + "."
+
+                report = {
+                    self.ugettext('Question'): sco_data.get(current_interaction_prefix + "description"),
+                    self.ugettext('Answer'): sco_data.get(current_interaction_prefix + "learner_response"),
+                    self.ugettext('Submissions count'): interactions_count
```
Comment (xitij2000) [inline scormxblock/scormxblock.py]: Note, "Submissions count" is something specific to the xblock-poll block.
This need not be included if not necessary or could be called "Interactions count" or anything else. 

Likewise, if there are other pieces of data about the student's interaction that could be of use, they could be added as well.

**Comment by xitij2000** [comment]
@moeez96 This can't be tested from Apros ATM. It needs a hawthorn or above devstack. The rough instructions for testing are:

- Install the xblock
- Setup the xblock with a scorm app
- Interact with the scorm block as a user
- Visit the instructor dashboard at http://localhost:18000/courses/{COURSE_ID}/instructor#view-data_download
- Select the new scorm block using "Select a section or problem" and click "Download a CSV of problem responses"
- The generated CSV should show up in the "Reports Available for Download" section. You might not be able to download it from there, but the report should be available in the lms container insider /edx/tmp-s3/...

**Comment by moeez96** [comment]
@xitij2000 Thank you for the instructions, I will set it up and test in practice.

File: scormxblock/scormxblock.py (hunk 1 above)
Comment (jawad-khan) [reply scormxblock/scormxblock.py]: Thanks for the insights. I'll update this accordingly.

**Comment by moeez96** [comment]
@xitij2000 I tested the PR by following the instructions [here](https://github.com/edx-solutions/edx-platform/pull/1344 ) at ginkgo devstack. The result looks correct except for one thing. 
If there are multiple entries against a single user in the database, the code keeps overwriting the same row in the report while populating it . This ends up with a single entry (the last one) to presist in the report and all previous entries are overwritten. I beleive the cause of this behaviour is mentioned [here](https://github.com/open-craft/xblock-poll/pull/50).
Can you let me know if the fix for this overwriting was made or is it planned anytime soon? Thanks.

**Comment by xitij2000** [comment]
@moeez96 Yes, I have make a PR for that: https://github.com/edx/edx-platform/pull/19507

However it needs some edX input, for how exactly to handle it. I am waiting for their feedback mainly. I have fixed the issue and bundled it with a couple of other enhancements. I will backport it to the solutions platform as well.

**Comment by xitij2000** [comment]
@moeez96 If you want you can test it with my PR which has the bugfix applied.

**Pull request merged**